pyinterprod/pronto/protein.py

```diff
@@ -289,3 +290,74 @@ def import_proteins(ora_url: str, pg_url: str):
 
     pg_con.close()
     logger.info("complete")
+
+
+def import_protein_pubmed(swp_url: str, pg_url: str):
+    logger.info("populating protein2publication")
+    pg_con = psycopg.connect(**url2dict(pg_url))
+    with pg_con.cursor() as pg_cur:
+        pg_cur.execute("DROP TABLE IF EXISTS protein2publication")
+        pg_cur.execute(
+            """
+            CREATE TABLE protein2publication (
+                PROTEIN_ACC VARCHAR(15) NOT NULL,
+                PUBMED_ID INTEGER NOT NULL
+            )
+            """
+        )
+
+        ora_con = oracledb.connect(swp_url)
+        ora_cur = ora_con.cursor()
+        swp2pmid = set()
+
+        ora_cur.execute(
+            """
+                SELECT E.ACCESSION, NVL(B.TEXT, SS.TEXT) AS TEXT
+                FROM SPTR.DBENTRY E
+                INNER JOIN SPTR.COMMENT_BLOCK B ON E.DBENTRY_ID = B.DBENTRY_ID
+                    AND B.COMMENT_TOPICS_ID = 2
+                INNER JOIN SPTR.CV_COMMENT_TOPICS CT 
+                    ON B.COMMENT_TOPICS_ID = CT.COMMENT_TOPICS_ID
+                LEFT OUTER JOIN SPTR.COMMENT_STRUCTURE S
+                    ON B.COMMENT_BLOCK_ID = S.COMMENT_BLOCK_ID
+                    AND S.CC_STRUCTURE_TYPE_ID = 1
+                LEFT OUTER JOIN SPTR.COMMENT_SUBSTRUCTURE SS
+                    ON S.COMMENT_STRUCTURE_ID = SS.COMMENT_STRUCTURE_ID
+                WHERE E.ENTRY_TYPE = 0
+                  AND E.MERGE_STATUS != 'R'           
+                  AND E.DELETED = 'N'
+            """
+        )
+        protein_text = {acc: text for acc, text in ora_cur.fetchall()}
+
+        sql = """
+            INSERT INTO protein2publication
+                (protein_acc, pubmed_id)
+            VALUES (%s, %s)
+        """
+        reg_pubmed = re.compile(r"(PubMed:(\d+))")
+        for acc, text in protein_text.items():
+            pmids = reg_pubmed.findall(text)
+            if pmids:
+                for pmid in pmids:
+                    swp2pmid.add((acc, pmid[1]))
```

Comment:
Explicit >> Implicit. Convert the string-representation of PMIDs to integers.
```suggestion
                    swp2pmid.add((acc, int(pmid[1])))
```